sycl/source/device_selector.cpp

```diff
@@ -28,8 +29,51 @@ static bool isDeviceOfPreferredSyclBe(const device &Device) {
          backend::level0;
 }
 
-device device_selector::select_device() const {
-  vector_class<device> devices = device::get_devices();
+// return a device with the requested deviceType (and requested backend)
+// if no such device is found, heuristic is used to select a device.
+// 'deviceType' is the desired device type
+//   info::device_type::all means it relies on the heuristic to select a device
+// 'be' is a specific desired GPU backend choice when multiple backends are
+// found.
```

Comment:
```suggestion
```

Reply:
I think this comment is important because 'be' optional parameter is only used for GPU backend choice. Other devices do not need backend choice.